### PR TITLE
Enhanced tests for interop / countrylist (Exposureapp-2733, EXPOSUREAPP-2744)

### PR DESCRIPTION
**Pull request by BMItr**
## Relates to
https://jira.itc.sap.com/browse/EXPOSUREAPP-2733
https://jira.itc.sap.com/browse/EXPOSUREAPP-2744

## Description
Added some basic test for InteroperabilityConfigurationFragmentViewModel

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        )
+
+        interoperabilityRepository.getAllCountries()
+        countryListLiveData.value = countryListFetched
+
+        val countryList =
+            interoperabilityConfigurationFragmentViewModel.countryList.getOrAwaitValue()
+
+        assertTrue(countryList.isNotEmpty())
+        assertEquals(countryList.size, countryListFetched.size)
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt:60]: isNotEmpty and size check is checking the same thing

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    @BeforeEach
+    fun setupFreshViewModel() {
+        every { interoperabilityRepository.countryList } returns countryListLiveData
+        interoperabilityConfigurationFragmentViewModel =
+            InteroperabilityConfigurationFragmentViewModel(interoperabilityRepository)
+    }
+
+    @Test
+    fun testNoFetch() {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt:38]: IMHO the "test" prefix is to be dropped as all functions are tests - no added value.
Moreover, the name of the text could describe in more detail what is being tested. E.g. "countryListIsEmptyIfRepositoryReturnsNoData" or alike

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        val countryList =
+            interoperabilityConfigurationFragmentViewModel.countryList.getOrAwaitValue()
+
+        assertTrue(countryList.isEmpty())
+    }
+
+    @Test
+    fun testFetch() {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt:46]: same as above

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            interoperabilityConfigurationFragmentViewModel.countryList.getOrAwaitValue()
+
+        assertTrue(countryList.isNotEmpty())
+        assertEquals(countryList.size, countryListFetched.size)
+        assertTrue(countryList == countryListFetched)
+        verify { interoperabilityRepository.getAllCountries() }
+    }
+
+    @Test
+    fun testBackPress() {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt:66]: same as above

**Review by ralfgehrer — DISMISSED** [review]
Added some comments - some of which are personal preference. Looks fine overall. Thanks for adding more test cases.

**Comment by BMItter** [comment]
@ralfgehrer thx for quick review and your thoughts. I 'll apply some of them later..

**Review by mertsafter — DISMISSED** [review]

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt (hunk 1 above)
Comment (BMItter) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/interoperability/InteroperabilityConfigurationFragmentViewModelTest.kt:60]: never say never :-) but I assert this is true!

**Review by ralfgehrer — APPROVED** [review]

**Review by harambasicluka — APPROVED** [review]

**Pull request merged**